src/utils.js

```diff
@@ -0,0 +1,19 @@
+const getData = async (url) => {
+  try {
+    const res = await fetch(url, {
+      method: "GET",
+      headers: { "Content-Type": "application/json" },
+    });
+
+    return await res.json();
+  } catch (error) {
+    console.log("Something went wrong!");
```

Comment:
create more function like getCategoryData, getProductData
and add their respective urls also, then make changes in other files also